modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentResourceImpl.java

```diff
@@ -517,57 +517,10 @@ public StructuredContent patchStructuredContent(
 		_validateContentFields(
 			structuredContent.getContentFields(), ddmStructure);
 
-		int expirationDateMonth = 0;
-		int expirationDateDay = 0;
-		int expirationDateYear = 0;
-		int expirationDateHour = 0;
-		int expirationDateMinute = 0;
+		boolean neverExpire = _isNeverExpire(structuredContent, journalArticle);
 
-		Boolean neverExpire = structuredContent.getNeverExpire();
-
-		if (neverExpire == null) {
-			if (journalArticle.getExpirationDate() == null) {
-				neverExpire = true;
-			}
-			else {
-				neverExpire = false;
-			}
-		}
-
-		if (!neverExpire) {
-			Date date = new Date();
-
-			Date dateExpired = structuredContent.getDateExpired();
-
-			if (dateExpired == null) {
-				dateExpired = journalArticle.getExpirationDate();
-			}
-
-			if (dateExpired == null) {
-				dateExpired = new Date(date.getTime() + Time.YEAR);
-			}
-
-			if (dateExpired.after(date)) {
-				Calendar expirationCal = CalendarFactoryUtil.getCalendar(
-					contextUser.getTimeZone());
-
-				expirationCal.setTime(dateExpired);
-
-				expirationDateMonth = expirationCal.get(Calendar.MONTH);
-				expirationDateDay = expirationCal.get(Calendar.DATE);
-				expirationDateYear = expirationCal.get(Calendar.YEAR);
-				expirationDateHour = expirationCal.get(Calendar.HOUR);
-				expirationDateMinute = expirationCal.get(Calendar.MINUTE);
-
-				if (expirationCal.get(Calendar.AM_PM) == Calendar.PM) {
-					expirationDateHour += 12;
-				}
-			}
-			else {
-				throw new BadRequestException(
-					"Please enter a valid expiration date");
-			}
-		}
+		int[] expirationDate = _getExpirationDate(
```

Comment:
Please use the same names as com/liferay/translation/internal/info/item/updater/JournalArticleInfoItemFieldValuesUpdater.java:171

```
		int[] expirationDateArray = _getDateArray(
			user, latestArticle.getExpirationDate());
```

Reply:
done